src/main/java/com/conveyal/datatools/manager/controllers/api/UserController.java

```diff
@@ -383,18 +495,38 @@ private static String getUserIdUrl(Request req) {
 
     /**
      * Safely parse the request body into a JsonNode.
-     *
-     * @param req The initating request that came into datatools-server
+     * @param req The initiating request that came into datatools-server
      */
     private static JsonNode parseJsonFromBody(Request req) {
+        return parseJsonFromString(req, req.body());
+    }
+
+    private static JsonNode parseJsonFromString(Request req, String json) {
         try {
-            return mapper.readTree(req.body());
+            return mapper.readTree(json);
         } catch (IOException e) {
             logMessageAndHalt(req, 400, "Failed to parse request body", e);
             return null;
         }
     }
 
+
+    /**
+     * Utility method to parse generic objects from JSON String and return as list.
+     * @param req The initiating request that came into datatools-server
+     * @param json The string representation of JSON to be parsed.
+     * @param clazz The class used to map each JSON object extracted.
+     */
+    private static <T> List<T> getPOJOFromJSONAsList(Request req, String json, Class<T> clazz) {
```

Comment:
Since this file is pretty long, I think all of these utilities might fit better in the JsonUtil file.

Reply:
Updated.